[BOLT][DWARF][NFC] Refactor GDB Index into a new file

sayhaan opened the pull request:
Create a new class and file for functions that update GDB index.

sayhaan committed:
Refactor GDBIndex into new file

Summary: 

Test Plan: 

Reviewers: 

Subscribers: 

Tasks: 

Tags: 


Differential Revision: https://phabricator.intern.facebook.com/D58156977

ayermolo commented:
Add [BOLT][DWARF] to title

maksfb commented:
> Add [BOLT][DWARF] to title

And NFC somewhere.

ayermolo commented on bolt/include/bolt/Core/GDBIndex.h:
```diff
@@ -0,0 +1,60 @@
+public:
+  GDBIndex(BinaryContext &BC) : BC(BC) {}
+
+  /// Adds an GDBIndexTUEntry if .gdb_index section exists.
```
Right now it always happens. Maybe in follow up commit let's change it.
Meanwhile change comment to reflect current behavior.

ayermolo commented on bolt/include/bolt/Core/GDBIndex.h:
```diff
@@ -0,0 +1,60 @@
+//===-- bolt/Core/GDBIndex.h - GDB Index support -------*- C++ -*-===//
```
can you format this comment section to 80 characters width

ayermolo commented on bolt/include/bolt/Core/GDBIndex.h:
```diff
@@ -0,0 +1,60 @@
+      CUOffsetMap &CUMap, uint32_t NumCUs,
+      std::unique_ptr<DebugARangesSectionWriter> &ARangesSectionWriter);
+
+  /// Returns all entries needed for Types CU list
```
Period

ayermolo commented on bolt/lib/Core/GDBIndex.cpp:
```diff
@@ -0,0 +1,182 @@
+using namespace llvm::bolt;
+using namespace llvm::support::endian;
+
+void GDBIndex::addGDBTypeUnitEntry(const GDBIndexTUEntry &Entry) {
```
Look like some code that checks the section got lost in refactor.

ayermolo commented on bolt/lib/Core/GDBIndex.cpp:
```diff
@@ -0,0 +1,182 @@
+//===- bolt/Rewrite/GDBIndex.cpp -------------------------------------===//
```
This isn't in Rewrite

sayhaan committed:
Formatting changes and readding code

Summary: 

Test Plan: 

Reviewers: 

Subscribers: 

Tasks: 

Tags: 


Differential Revision: https://phabricator.intern.facebook.com/D58193743

sayhaan committed:
Formatting changes

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

ayermolo commented on bolt/include/bolt/Core/GDBIndex.h:
```diff
@@ -0,0 +1,62 @@
+public:
+  GDBIndex(BinaryContext &BC) : BC(BC) {}
+
+  std::mutex DWARFRewriterMutex;
```
Rename to more relevant variable.

sayhaan committed:
Make variable name more relevant

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

ayermolo commented on bolt/lib/Core/GDBIndex.cpp:
```diff
@@ -0,0 +1,185 @@
+using namespace llvm::bolt;
+using namespace llvm::support::endian;
+
+void GDBIndex::addGDBTypeUnitEntry(const GDBIndexTUEntry &Entry) {
```
why was this changed to move by reference/push_bcack from &&/emplace_back?

ayermolo commented on bolt/lib/Core/GDBIndex.cpp:
```diff
@@ -0,0 +1,185 @@
+
+void GDBIndex::updateGdbIndexSection(
+    CUOffsetMap &CUMap, uint32_t NumCUs,
+    std::unique_ptr<DebugARangesSectionWriter> &ARangesSectionWriter) {
```
Change this to DebugARangesSectionWriter &ARangesSectionWriter

sayhaan committed:
Refactoring functions

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

sayhaan committed:
Formatting changes

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

sayhaan committed:
Refactoring functions

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
